spec/index.adoc

```diff
@@ -619,29 +627,97 @@ See <<state-tree>> for details on the state tree.
+Canister methods that do not change the canister state can be executed more efficiently.
+A query call provides that ability: it is a fast, but less secure way to call a canister.
+Only methods that are explicitly marked as “query methods” by the canister can be called this way.
 
-In order to make a query call to canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/query`. The request body consists of an authentication envelope with a `content` map with the following fields:
+In order to make a query call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/query`.
+The request body consists of an authentication envelope with a `content` map with the following fields:
 
-* `request_type` (`text`): Always `query`
+* `request_type` (`text`): Always `"query"`
 * `sender`, `nonce`, `ingress_expiry`: See <<authentication>>
 * `canister_id` (`blob`): The principal of the canister to call.
-* `method_name` (`text`): Name of the canister method to call
-* `arg` (`blob`): Argument to pass to the canister method
+* `method_name` (`text`): Name of the canister method to call.
+* `arg` (`blob`): Argument to pass to the canister method.
 
-If the call resulted in a reply, the response is a CBOR (see <<cbor>>) map with the following fields:
+If the query call resulted in a reply, the response is a CBOR (see <<cbor>>) map with the following fields:
 
-* `status` (`text`): `replied`
+* `status` (`text`): `"replied"`
 * `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data.
 
 If the call resulted in a reject, the response is a CBOR map with the following fields:
 
-* `status` (`text`): `rejected`
+* `status` (`text`): `"rejected"`
 * `reject_code` (`nat`): The reject code (see <<reject-codes>>).
 * `reject_message` (`text`): a textual diagnostic message.
-* `error_code` (text): an optional implementation-specific textual error code (see <<error-codes>>).
+* `error_code` (`text`): an optional implementation-specific textual error code (see <<error-codes>>).
+
+[#http-signed-query]
+=== Request: Signed query call
+
+NOTE: The signed query call API is considered EXPERIMENTAL. Canister developers must be aware that the API may evolve in a non-backward-compatible way.
+
+A *signed* query call is a query call amended with signatures produced by IC nodes evaluating the query call to certify the query response.
+
+In order to make a signed query call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/signed_query`.
+The request body has the same structure as the request body for <<http-query>> with the `content` map having the following *additional* fields:
+
+* (optional) `return_signatures` (`text`): Determines whether and how many signatures are returned in the response; possible values are `"none"`, `"one"`.
```

Comment:
> why the fields are text instead of nat ?

nat could suggest you can specify the exact number which won't be the case in the future when we want to provide multiple signatures (where the exact number will depend on the subnet size and the user won't be able to provide the exact number of signatures to retrieve)

> maybe requested_signatures, requested_certifications is better naming?

sounds good to me; please put "thump up" if you like "requested_signatures", "requested_certifications"

Reply:
renamed the fields